lib/beaker/dsl/install_utils.rb

```diff
@@ -701,6 +701,56 @@ def add_system32_hosts_entry(host, opts = {})
         end
       end
 
+      # Configure puppet.conf on the given host based upon a provided hash
+      # @param [Host] host The host to configure puppet.conf on
+      # @param [Hash{Symbol=>String}] opts
+      # @option opts [Hash{String=>String}] :main configure the main section of puppet.conf
+      # @option opts [Hash{String=>String}] :agent configure the agent section of puppet.conf
+      #
+      # @example will configure /etc/puppet.conf on the puppet master.
+      #   config = {
+      #     'main' => {
+      #       'server'   => 'testbox.test.local',
+      #       'certname' => 'testbox.test.local',
+      #       'logdir'   => '/var/log/puppet',
+      #       'vardir'   => '/var/lib/puppet',
+      #       'ssldir'   => '/var/lib/puppet/ssl',
+      #       'rundir'   => '/var/run/puppet'
+      #     },
+      #     'agent' => {
+      #       'environment' => 'dev'
+      #     }
+      #   }
+      #   configure_puppet(master, config)
+      #
+      # @api dsl
+      # @return nil
+      def configure_puppet(host, opts = {})
+        if host['platform'] =~ /windows/
```

Comment:
Final nit, this won't work with multi-host.  Is that the desired state?  Any chance that you would just run this on all hosts?